tunnel/src/main/java/com/wireguard/config/Interface.java

```diff
@@ -231,6 +277,14 @@ public String toWgQuickString() {
         listenPort.ifPresent(lp -> sb.append("ListenPort = ").append(lp).append('\n'));
         mtu.ifPresent(m -> sb.append("MTU = ").append(m).append('\n'));
         sb.append("PrivateKey = ").append(keyPair.getPrivateKey().toBase64()).append('\n');
+        for (final String script : preUp)
+            sb.append("PreUp = ").append(script).append('\n');
+        for (final String script : postUp)
+            sb.append("PostUp = ").append(script).append('\n');
+        for (final String script : preDown)
+            sb.append("PreDown = ").append(script).append('\n');
+        for (final String script : postDown)
+            sb.append("PostDown = ").append(script).append('\n');
```

Comment:
This fails since Android's `wg-quick` implementation doesn't support these actions. I have some changes pending that add support there as well but I'll wait to send that patch until the outcome of the potential redesign is completed.